homeassistant/components/simplisafe/__init__.py

```diff
@@ -517,11 +519,28 @@ def _async_process_new_notifications(self, system: SystemType) -> None:
 
         self._system_notifications[system.system_id] = latest_notifications
 
-    async def _async_websocket_on_connect(self) -> None:
+    async def _async_websocket_connect(self) -> None:
         """Define a callback for connecting to the websocket."""
         if TYPE_CHECKING:
             assert self._api.websocket
-        await self._api.websocket.async_listen()
+
+        should_reconnect = True
+
+        try:
+            await self._api.websocket.async_connect()
+            await self._api.websocket.async_listen()
+        except asyncio.CancelledError:
+            should_reconnect = False
```

Comment:
Does it make sense to `raise` here instead to let the cancelation propagate?

Reply:
The only way we cancel is if HASS is shutting down (I followed a similar pattern that Z-Wave JS uses). That doesn't seem like a raise, unless I'm missing something?

Reply:
https://docs.python.org/3.8/library/asyncio-exceptions.html#asyncio.CancelledError

This exception can be caught to perform custom operations when asyncio Tasks are cancelled. **In almost all situations the exception must be re-raised.**

If its not re-raised it will prevent cancelation.

Reply:
Meaning re-raise `CancelledError`? I've always struggled with those docs. 😂